app/models/manageiq/providers/base_manager/event_catcher/runner.rb

```diff
@@ -80,12 +80,21 @@ def stop_event_monitor
     raise NotImplementedError, "must be implemented in subclass"
   end
 
+  # the monitor runs in it's own thread and only yields raw events to put them into an internal queue
+  # this queue is read from another thread which parses the events and queues them for writing
+  # @yield [event_or_events] any object or array of objects to be parsed by parse_event
   def monitor_events
     raise NotImplementedError, "must be implemented in subclass"
   end
 
-  def process_event(_event)
-    raise NotImplementedError, "must be implemented in subclass"
+  def event_parser
+    @ems.class::EventParser
+  end
+
+  def process_event(event)
+    event_hash = event_parser.event_to_hash(event, @cfg[:ems_id])
```

Comment:
@matthewd how do you like this? Now `EmsEvent` does not know anything about an `EventParser` and all parsing logic can move to the `EventParser`. The `EventCatcher::Runner` subclass does not have to implement `process_event` anymore